bluepysnap/network.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) 2020-2021, EPFL/Blue Brain Project
+
+# This file is part of BlueBrain SNAP library <https://github.com/BlueBrain/snap>
+
+# This library is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License version 3.0 as published
+# by the Free Software Foundation.
+
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""Module containing the Abstract classes for the Network."""
+import abc
+
+import pandas as pd
+import numpy as np
+from cached_property import cached_property
+
+
+from bluepysnap import utils
+from bluepysnap.exceptions import BluepySnapError
+
+
+class NetworkObject(abc.ABC):
+    """Abstract class for the top level NetworkObjects accessor."""
+
+    def __init__(self, circuit):
+        """Initialize the top level NetworkObjects accessor."""
+        self._circuit = circuit
+
+    def _get_populations(self, cls, config):
```

Comment:
If I understand correctly, this function will always be called with cls=EdgeStorage when self is  of type Edges and cls=NodeStorage when self is of type Nodes.

In this case, why not having a class level attribute Storage in the NetworkObject class that you set to 
Storage=EdgeStorage in Edges
Storage=NodeStorage in Nodes

Then in the following code you can remove cls from the argument list and replace cls by Storage in the body.
Conceptually I think it also makes it more explicit that Edges is tied to EdgeStorage the same way Nodes is tied to NodeStorage.

Reply:
The rational being this function, compared to use a class object or something similar, was the single use of the Storage in the whole class inside _get_populations (which is not public). Plus I will need to define the config and so to override the _get_population anyway or to add a Class object / Class member 
So I think it is more convoluted this way and I don t see the possible usage of a : 
Nodes.Storage or Edges.Storage for the user.

Reply:
I'm good with this. Thanks